tests/test_url.py

```diff
@@ -1170,3 +1171,20 @@ def test_requoting():
     u = URL("http://127.0.0.1/?next=http%3A//example.com/")
     assert u.raw_query_string == "next=http://example.com/"
     assert str(u) == "http://127.0.0.1/?next=http://example.com/"
+
+def test_idna_decode_cache():
```

Comment:
How IDNA correlates with IP `127.0.0.1`?

Reply:
This test is only for the cache stuff, idna tests are up

Reply:
Yeap. I just stumbled upon test name. May be remove idna from it name to prevent confusion? Cache feature applies for all kind of hostname.

Reply:
I'd rename this to `test_decode_host_cache`, and `test_encode_host_cache` below. Easier to correlate with the actual function names they cover.

Reply:
The main reason of caching is for usage of idna, @asvetlov what do you think about renaming this test something like `test_hostname_encoding_cache`?

Reply:
Forget my last comment, I haven't had read the @mjpieters comment, I'll rename like it